python/pyspark/sql/tests/connect/client/test_client.py

```diff
@@ -159,6 +159,11 @@ def Interrupt(self, req: proto.InterruptRequest, metadata):
             resp.session_id = self._session_id
             return resp
 
+    # The _cleanup_ml_cache invocation will hang in this test (no valid spark cluster)
+    # and it blocks the test process exiting because it is registered as the atexit handler
+    # in `SparkConnectClient` constructor. To bypass the issue, patch the method in the test.
+    SparkConnectClient._cleanup_ml_cache = lambda _: None
```

Comment:
can we just del this `axexit` as well?
ml cache is always cleaned when the SessionHolder is closed in the server side.

Reply:
Is the session ensured to be closed when the client exits or client crashes ? @HyukjinKwon

Reply:
I think we should do it in follow-up PR, it needs some discussion.

Reply:
> Is the session ensured to be closed when the client exits or client crashes ?

From what I know, I don't think so